src/psyclone/tests/transformations_test.py

```diff
@@ -221,3 +219,126 @@ def test_regiontrans_wrong_children():
         RegionTrans.validate(rtrans, parent.children)
     assert ("Cannot apply a transformation to multiple nodes when one or more "
             "is a Schedule" in str(err.value))
+
+
+def test_regiontrans_wrong_options():
+    '''Check that the validate method raises the expected error if passed
+        options that are not contained in a dictionary.
+
+    '''
+    # RegionTrans is abstract so use a concrete sub-class
+    region_trans = ACCParallelTrans()
+    with pytest.raises(TransformationError) as excinfo:
+        RegionTrans.validate(region_trans, None, options="invalid")
+    assert ("Transformation apply method options argument must be a "
+            "dictionary but found 'str'." in str(excinfo.value))
+
+# Tests for ProfileRegionTrans
+
+
+def dummy_nodes():
+    '''Utility routine which create a minimal invoke and schedule
+    hierarchy for use in subsequent ProfileRegionTrans tests.
+
+    :returns: a list containing a single node connected to a minimal \
+    InvokeSchedule which is connected to a minimal Schedule. This \
+    hierarchy is required for code unrelated to the tests.
+    :rtype: list of :py:class:`psyclone.psyGen.Node`
+
+    '''
+    schedule = InvokeSchedule(None, None)
+    schedule.invoke = Invoke(None, None, None)
+    node = Node()
+    node.parent = schedule
+    nodes = [node]
+    schedule.children = nodes
+    return nodes
+
+
+def test_profilerregiontrans_noname(monkeypatch):
+    '''Check that when no name is supplied to the profile transformation
+    then it creates a profile node with no name information.
+
+    '''
+    class Dummy(object):
+        '''Dummy class that checks the name argument is None.
+
+        :param NonType parent: dummy to conform to ProfileNode arguments.
+        :param NonType children: dummy to conform to ProfileNode arguments.
+        :param NoneType name: the name argument that we are testing.
+
+        '''
+        def __init__(self, parent=None, children=None, name=None):
+            self._parent = parent
+            self._children = children
+            if name is None:
+                raise Exception("test passed")
+    monkeypatch.setattr("psyclone.profiler.ProfileNode", Dummy)
+    # Create and apply the transformation.
+    profile_trans = ProfileRegionTrans()
+    with pytest.raises(Exception) as excinfo:
+        _ = profile_trans.apply(dummy_nodes())
+    assert "test passed" in str(excinfo.value)
```

Comment:
This all seems a bit long-winded. Couldn't you just apply the transform as normal and then check that the ProfileNode._*name properties are None?

Reply:
I rather liked it. Anyway, I've changed the tests and the examples are now much shorter and clearer, so you were right.

Reply:
Thanks.